src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseFilesService.cs

```diff
@@ -89,27 +112,41 @@ public class ReleaseFilesService : IReleaseFilesService
                             await CreateOrUpdateFileReference(metadataFile.FileName.ToLower(), releaseId,ReleaseFileTypes.Metadata);
                             await _context.SaveChangesAsync();
-                            await UploadFileToStorageAsync(blobContainer, releaseId, dataFile, ReleaseFileTypes.Data, dataInfo);
-                            await UploadFileToStorageAsync(blobContainer, releaseId, metadataFile,ReleaseFileTypes.Metadata, metaDataInfo);
+
+                            var dataInfo = GetDataFileMetaValues(
+                                name: name,
+                                metaFileName: metadataFile.FileName,
+                                userName: userName,
+                                numberOfRows:  CalculateNumberOfRows(dataFile.OpenReadStream())
+                            );
+                            var metaDataInfo = GetMetaDataFileMetaValues(
+                                dataFileName: dataFile.FileName,
+                                userName: userName,
+                                numberOfRows: CalculateNumberOfRows(metadataFile.OpenReadStream())
+                            );
+
+                            await UploadFileToStorage(releaseId, dataFile, ReleaseFileTypes.Data, dataInfo);
+                            await UploadFileToStorage(releaseId, metadataFile,ReleaseFileTypes.Metadata, metaDataInfo);
+
                             await _importService.Import(releaseId, dataFile.FileName.ToLower(),
                                 metadataFile.FileName.ToLower(), dataFile, false);
 
-                            var file = blobContainer.GetBlockBlobReference(
+                            var blob = await _blobStorageService.GetBlob(
+                                PrivateFilesContainerName,
                                 AdminReleasePathWithFileReference(fileReference)
                             );
-                            await file.FetchAttributesAsync();
 
                             return new DataFileInfo
                             {
                                 Id = fileReference.Id,
-                                Extension = GetExtension(file),
-                                Name = GetName(file),
-                                Path = file.Name,
-                                Size = GetSize(file),
-                                MetaFileName = GetMetaFileName(file),
-                                Rows = GetNumberOfRows(file),
-                                UserName = GetUserName(file),
-                                Created = file.Properties.Created
+                                Extension = blob.Extension,
+                                Name = blob.Name,
+                                Path = blob.Path,
+                                Size = blob.Size,
+                                MetaFileName = blob.GetMetaFileName(),
```

Comment:
I'm not sure  why some of these properties in DataFileInfo are marked as being optional. Maybe we can clean them up here at the same time.

Reply:
Now marked these as required